server/src/main/java/org/elasticsearch/index/query/PrefixQueryBuilder.java

```diff
@@ -61,6 +63,9 @@ public PrefixQueryBuilder(String fieldName, String value) {
         if (value == null) {
             throw new IllegalArgumentException("value cannot be null");
         }
+        if (value.length() > IndexSettings.MAX_REGEX_LENGTH_SETTING.get(Settings.EMPTY)) {
```

Comment:
This is incorrect as it will always use the default value of `1000` regardless how the user change this setting. 

Please follow `RegexpQueryBuilder.java` how this setting should be used: `context.getIndexSettings().getMaxRegexLength()`.

Please also follow the standard of a good error message as in `RegexpQueryBuilder.java`